src/lib/LBT/LBT.cpp

```diff
@@ -0,0 +1,111 @@
+  // SF9 ~218us (Odd one out, measured to same as SF7 wait time)
+
+  switch(SF)
+    {
+      case SX1280_LORA_SF5: return 100;
+      case SX1280_LORA_SF6: return 141;
+      case SX1280_LORA_SF7: return 218;
+      case SX1280_LORA_SF9: return 218;
+      default: return 218; // Values above 100mW are not relevant, default to 100mW threshold
+    }
+}
+
+int8_t ICACHE_RAM_ATTR PowerEnumToLBTLimit(PowerLevels_e txPower)
+{
+  // Calculated from EN 300 328, adjusted for 800kHz BW for sx1280
+  // TL = -70 dBm/MHz + 10*log10(0.8MHz) + 10 × log10 (100 mW / Pout) (Pout in mW e.i.r.p.)
+  // This threshold should be offset with a #define config for each HW that corrects for antenna gain,
+  // different RF frontends.
+  // TODO: Maybe individual adjustment offset for differences in
+  // rssi reading between bandwidth setting is also necessary when other BW than 0.8MHz are used.
+
+  switch(txPower)
+  {
+    case PWR_10mW: return -61 + LBT_RSSI_THRESHOLD_OFFSET_DB;
+    case PWR_25mW: return -65 + LBT_RSSI_THRESHOLD_OFFSET_DB;
+    case PWR_50mW: return -68 + LBT_RSSI_THRESHOLD_OFFSET_DB;
+    case PWR_100mW: return -71 + LBT_RSSI_THRESHOLD_OFFSET_DB;
+    // Values above 100mW are not relevant, default to 100mW threshold
+    default: return -71 + LBT_RSSI_THRESHOLD_OFFSET_DB;
+  }
+}
+
+void ICACHE_RAM_ATTR BeginClearChannelAssessment(void)
+{
+  // Listen Before Talk (LBT) aka clear channel assessment (CCA)
+  // Not interested in packets or interrupts while measuring RF energy on channel.
+  Radio.SetDioIrqParams(SX1280_IRQ_RADIO_NONE, SX1280_IRQ_RADIO_NONE, SX1280_IRQ_RADIO_NONE, SX1280_IRQ_RADIO_NONE);
+  if(Radio.GetMode() != SX1280_MODE_RX)
+  {
+    Radio.RXnb();
+    rxStartTime = micros();
+  }
+}
+
+bool ICACHE_RAM_ATTR ChannelIsClear(void)
+{ 
+  LBTSuccessCalc.inc(); // Increment count for every channel check
+
+  // Read rssi after waiting the minimum RSSI valid delay.
+  // If this function is called long enough after RX enable,
+  // this will always be ok on first try as is the case for TX.
+
+  // TODO: Better way than busypolling this for RX?
+  // this loop should only run for RX, where listen before talk RX is started right after FHSS hop
+  // so there is no dead-time to run RX before telemetry TX is supposed to happen.
+  // if flipping the logic, so telemetry TX happens right before FHSS hop, then TX-side ends up with polling here instead?
+  // Maybe it could be skipped if there was only TX of telemetry happening when FHSShop does not happen.
+  // Then RX for LBT could stay enabled from last received packet, and RSSI would be valid instantly.
+  // But for now, FHSShops and telemetry rates does not divide evenly, so telemetry will some times happen
+  // right after FHSS and we need wait here.
+
+  uint32_t validRSSIdelayUs = SpreadingFactorToRSSIvalidDelayUs((SX1280_RadioLoRaSpreadingFactors_t)ExpressLRS_currAirRate_Modparams->sf);
+
+  while(micros() - rxStartTime < validRSSIdelayUs);
```

Comment:
Should this be using `delayMicroseconds` instead?

Reply:
with delayMicroseconds() the total waiting time since RXenable becomes longer than necessary. I did it this way to make the wait as close as possible to the necessary wait-time. For the code running on TX, this while-loop doesn't loop at all, since the wait-time has already been waited while other code was running.

Reply:
I don't see how using delayMicroSeconds like this is going to make it delay longer than necessary, unless delayMicroSeconds is not accurate. But, if as you say it's mostly a no-op, then perhaps a comment to that effect.
```cpp
int delay = validRSSIdelayUs - (micros() - rxStartTime);
if (delay>0)
{
    delayMicroSeconds(delay);
}
```

Reply:
Yeah, this could work, could you explain the benefit of using delayUs? Is it non-blocking? Does the framework support multiple levels of interrupts?

Reply:
It is also just a busy loop as far as I can ascertain, but it makes the intent of the code clearer.

Reply:
I do agree, will implement